frontend/src/features/recommendation/RecommendationApi.tsx

```diff
@@ -74,31 +78,60 @@ const getMetadataFilter = (
   return metadata;
 };
 
+export enum ScoreModeEnum {
+  DEFAULT = 'default',
+  ALL_EQUAL = 'all_equal',
+  TRUSTED_ONLY = 'trusted_only',
+}
+
 /**
  * Return the recommendations of a given poll.
  */
 export const getRecommendations = async (
   pollName: string,
   limit: number,
   searchString: string,
-  criterias: PollCriteria[]
+  criterias: PollCriteria[],
+  pollOptions?: SelectablePoll
 ): Promise<PaginatedRecommendationList> => {
   const params = new URLSearchParams(searchString);
 
   buildDateURLParameter(pollName, params);
 
+  let criteriaWeights = Object.fromEntries(
+    criterias.map((c) => [c.name, getParamValueAsNumber(params, c.name)])
+  );
+
+  /*
+    Temporary workaround for "presidentielle2022":
+    Contrary to poll "videos" where the default order for recommendations is an aggregation of
+    criteria, here we need to sort by the main criterion only (if custom weights are not provided).
+    2 things to fix before removing this workaround:
+      - update the backend implementation of "tournesol_score" to apply this new definition
+      - make sure that `CriteriaFilter` behaves correctly on initialization, and that its state
+      persists after the filter section is closed or the page is refreshed.
+  */
+  if (
+    pollName === PRESIDENTIELLE_2022_POLL_NAME &&
+    Object.values(criteriaWeights).filter((x) => x != null).length === 0
+  ) {
+    criteriaWeights = {
+      ...Object.fromEntries(criterias.map((c) => [c.name, 0])),
+      [pollOptions?.mainCriterionName ?? '']: 100,
+    };
+  }
+
   try {
     return await PollsService.pollsRecommendationsList({
       name: pollName,
       limit: limit,
       offset: getParamValueAsNumber(params, 'offset'),
       search: params.get('search') ?? undefined,
       dateGte: params.get('date_gte') ?? undefined,
-      unsafe: params.get('unsafe') === 'true' ?? undefined,
+      unsafe: params.get('unsafe') === 'true' || pollOptions?.unsafeDefault,
```

Comment:
I don't think this is correct. The way it is written, I believe that if a poll shows unsafe by default it becomes impossible to request safe.

```suggestion
      unsafe: params.has('unsafe') ? params.get('unsafe') === 'true' : pollOptions?.unsafeDefault,
```

Reply:
Fixed in https://github.com/tournesol-app/tournesol/pull/882/commits/afa7e75742e4c76ac994cccf33b0f11935ca78a4 (with linter applied)